### ISSUE TITLE
Auto/BabyAGI Programatic access

### ISSUE THREAD

**Issue by Kav-K** [run-llama/llama-lab#9]
Any plans in the future for enabling "calling" autoGPT/babyAGI through an endpoint or a function so that we can receive results in tidbits that can be used programatically, like doing a .query() call, kinda?

**Comment by logan-markewich** [comment]
Interesting idea.

So you mean, letting the AGI run off in it's loop in the background, and then pulling in it's status or results as it goes?

At least in llama_agi, all the task history (current and completed) is stored in two GPTListIndex instances, so querying against it would be easy enough.

**Comment by Kav-K** [comment]
I was thinking more so, a sequential API that allows us to see the response of each step and then send a request to proceed it as well

**Comment by Kav-K** [comment]
Yeah! And I feel like it'd be a useful features that would allow a lot of platforms to easily run AutoGPT as a service! :)
